testing/deployer/sprawl/catalog.go

```diff
@@ -301,11 +471,179 @@ RETRY:
 	return nil
 }
 
+type Resource[V proto.Message] struct {
+	Resource *pbresource.Resource
+	Data     V
+}
+
+func (r *Resource[V]) Build() (*pbresource.Resource, error) {
+	anyData, err := anypb.New(r.Data)
+	if err != nil {
+		return nil, err
+	}
+	r.Resource.Data = anyData
+	return r.Resource, nil
+}
+
+type ServiceResources struct {
+	Workload         *Resource[*pbcatalog.Workload]
+	HealthStatuses   []*Resource[*pbcatalog.HealthStatus]
+	Destinations     *Resource[*pbmesh.Destinations]
+	WorkloadIdentity *Resource[*pbauth.WorkloadIdentity]
+}
+
+func serviceInstanceToResources(
+	cluster *topology.Cluster,
+	node *topology.Node,
+	svc *topology.Service,
+) *ServiceResources {
+	if svc.IsMeshGateway {
+		panic("v2 does not yet support mesh gateways")
+	}
+
+	tenancy := &pbresource.Tenancy{
+		Partition: svc.ID.Partition,
+		Namespace: svc.ID.Namespace,
+	}
+
+	var (
+		wlPorts  = map[string]*pbcatalog.WorkloadPort{}
+		svcPorts []*pbcatalog.ServicePort
+	)
+	for name, port := range svc.Ports {
+		wlPorts[name] = &pbcatalog.WorkloadPort{
+			Port:     uint32(port),
+			Protocol: pbcatalog.Protocol_PROTOCOL_TCP,
+		}
+
+		svcPorts = append(svcPorts, &pbcatalog.ServicePort{
+			TargetPort: name,
+			Protocol:   pbcatalog.Protocol_PROTOCOL_TCP,
+		})
+	}
+
+	var (
+		selector = &pbcatalog.WorkloadSelector{
+			Names: []string{svc.Workload},
+		}
+
+		workloadRes = &Resource[*pbcatalog.Workload]{
+			Resource: &pbresource.Resource{
+				Id: &pbresource.ID{
+					Type:    pbcatalog.WorkloadType,
+					Name:    svc.Workload,
+					Tenancy: tenancy,
+				},
+				Metadata: svc.Meta,
+			},
+			Data: &pbcatalog.Workload{
+				// NodeName: node.PodName(),
```

Comment:
Leaving the node stuff commented out for now until nodes can properly handle tenancy.

Reply:
Nodes properly handling tenancy will come later.

Reply:
Can you put a `// TODO` or something in here?